gradio/components/gallery.py

```diff
@@ -155,13 +158,21 @@ def postprocess(
                 file_path = str(utils.abspath(file))
             elif isinstance(img, str):
                 file_path = img
-                url = img if is_http_url_like(img) else None
+                if is_http_url_like(img):
+                    orig_name = Path(urlparse(img).path).name
+                else:
+                    orig_name = Path(img).name
             elif isinstance(img, Path):
                 file_path = str(img)
+                orig_name = img.name
             else:
                 raise ValueError(f"Cannot process type as image: {type(img)}")
             entry = GalleryImage(
-                image=FileData(path=file_path, url=url), caption=caption
+                # Leave `url` as None so it will be replaced with a local cache path during postprocessing (https://github.com/gradio-app/gradio/blob/c5ea13bdf7363224863760e53fbcdbed81754628/gradio/blocks.py#L1421)
```

Comment:
This will introduce a regression in performance when compared to main.

You can check with the `gallery_component_events` demo. The postprocess will take several seconds as opposed to < 1 second.

### This change
![gallery_postprocess_pr](https://github.com/gradio-app/gradio/assets/41651716/f77ab61f-fc3f-4a90-a425-4b20799fe4f4)

### Main
![gallery_postprocess_main](https://github.com/gradio-app/gradio/assets/41651716/9b494f64-2151-4cef-99e2-b47ed7048483)

In the past, people have used gr.Gallery to store many image URLs so I'm worried about degrading the experience for them.

Reply:
That's an important point, thanks.
I understand `gr.Gallery()` is not just a group of `gr.Images()` but the quantity matters.
So I'm gonna go back to the previous path and improve it.